beeline.go

```diff
@@ -96,16 +99,51 @@ type Config struct {
 	PprofTagging bool
 }
 
+func IsClassicKey(config Config) bool {
+	// classic key has 32 characters
+	return len(config.WriteKey) == 32
+}
+
 // Init intializes the honeycomb instrumentation library.
 func Init(config Config) {
 	userAgentAddition := fmt.Sprintf("beeline/%s", version)
 
 	if config.WriteKey == "" {
 		config.WriteKey = defaultWriteKey
+		fmt.Println("WARN: Missing API Key.")
 	}
-	if config.Dataset == "" {
-		config.Dataset = defaultDataset
+
+	if config.ServiceName == "" {
+		fmt.Println("WARN: Missing service name.")
+	}
+
+	if IsClassicKey(config) {
+		// if classic and missing dataset, warn on that
+		if config.Dataset == "" {
+			config.Dataset = defaultDataset
+			fmt.Println("WARN: Missing dataset. Data will be sent to:", defaultDataset)
+		}
+	} else {
+		// set default service name if not provided
+		if config.ServiceName == "" {
+			config.ServiceName = defaultServiceName
+			// try to append default with process name
+			if executable, err := os.Executable(); err == nil {
+				config.ServiceName = defaultServiceName + ":" + filepath.Base(executable)
+			} else {
+				// fall back to language if process name is unavailable
+				config.ServiceName = defaultServiceName + ":go"
+			}
+		}
+
+		// non classic key will ignore dataset, warn if configured
+		if config.Dataset != "" {
+			fmt.Println("WARN: Dataset is ignored in favor of service name. Data will be sent to service name:", config.ServiceName)
+		}
+		// but set dataset based on service name
+		config.Dataset = config.ServiceName
```

Comment:
This needs to check if it's unknown service and if so, truncate to `unknown_service` before sending.